### DISCUSSION TITLE
Support Google Play track promotion

### DISCUSSION THREAD

**Discussion by chris-wickens** [codemagic-ci-cd/codemagic-docs#1966] [Ideas]
### Use case

As far as I can tell it's only possible to publish an Android build to a single track in the same workflow. I need a build to publish to the internal track for immediately testable builds, and then to auto-promote the build to the closed testing track so it's available to a wider Google Group as soon as it passes Google Play review. This latter step I currently need to perform manually.

One potential workaround might be to set up fastlane and call their Track Promotion action directly, but obviously that's not ideal as we'd have to do the Google Play setup twice. Another would be to run multiple workflows with a different track specified each time, but I have separate workflows for each flavor already and can't afford to double them again. :(

### Idea

Introduce a mechanism to auto-promote a build to later stage tracks, similar to fastlane's implementation: http://docs.fastlane.tools/actions/supply/#track-promotion

**Comment by mikhail-tokarev** [comment]
Hi @chris-wickens 

thank you for this idea. We have added in to our backlog. Can you also share if you use Flutter Workflow Editor or `codemagic.yaml` file for your builds?

**Reply by chris-wickens** [reply]
Thanks @mikhail-tokarev. I've just made the switch to codemagic.yaml as I needed melos to test sub-packages. Prior to that the workflow editor was working great for us, so track promotion would have been useful to have in the editor as well.

**Comment by helinanever** [comment]
Hi @chris-wickens, this feature is now available in Codemagic. To use it, you need to update your Google Play configuration in codemagic.yaml with an additional subsection called `release_promotion`, see the docs [here](https://docs.codemagic.io/yaml-publishing/google-play/#configure-publishing-in-codemagicyaml).
